lib/folio_client/authenticator.rb

```diff
@@ -3,24 +3,42 @@
 class FolioClient
   # Fetch a token from the Folio API using login_params
   class Authenticator
-    def self.token(login_params, connection)
-      new(login_params, connection).token
+    def self.token(login_params, connection, legacy_auth, cookie_jar)
+      new(login_params, connection, legacy_auth, cookie_jar).token
     end
 
-    def initialize(login_params, connection)
+    def initialize(login_params, connection, legacy_auth, cookie_jar)
       @login_params = login_params
       @connection = connection
+      @legacy_auth = legacy_auth
+      @cookie_jar = cookie_jar
     end
 
     # Request an access_token
+    # rubocop:disable Metrics/AbcSize
     def token
-      response = connection.post('/authn/login', login_params.to_json)
+      response = connection.post(login_endpoint, login_params.to_json)
 
       UnexpectedResponse.call(response) unless response.success?
 
-      JSON.parse(response.body)['okapiToken']
+      # remove legacy_auth once new tokens enabled on Poppy
+      if legacy_auth
+        JSON.parse(response.body)['okapiToken']
+      else
+        access_cookies = cookie_jar.cookies.select { |cookie| cookie.name == 'folioAccessToken' }
+        access_cookies[0].value
```

Comment:
Since we only care about the first value, we could consider `Enumerable#find` here too:

```suggestion
        cookie_jar.cookies.find { |cookie| cookie.name == 'folioAccessToken' }.value
```